File: app/src/main/java/br/com/hussan/githubapi/AppComponent.java
package br.com.hussan.githubapi;

import javax.inject.Singleton;

import br.com.hussan.githubapi.data.DataModule;
import br.com.hussan.githubapi.ui.repositories.MainActivity;
import dagger.Component;

/**
 * Created by hussan on 10/10/16.
 */
@Singleton
@Component(modules = {AppModule.class, DataModule.class})
public interface AppComponent {
    void inject(MainActivity activity);
}

File: app/src/main/kotlin/br/com/hussan/githubapi/AppComponent.kt
package br.com.hussan.githubapi

import br.com.hussan.githubapi.data.api.ApiModule
import br.com.hussan.githubapi.injection.ui.PresenterModule
import br.com.hussan.githubapi.injection.ui.UiComponent
import dagger.Component
import javax.inject.Singleton

/**
 * Created by hussan on 10/10/16.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class, ApiModule::class))
interface AppComponent {
    fun plus(presenterModule: PresenterModule) : UiComponent
}
